### PR TITLE
Add middleware for Prometheus Exporter

### PR DESCRIPTION
**Pull request by hananiel**
@cijothomas  wanted to start this conversation sooner than later.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            foreach (var metric in exporter.GetAndClearLongMetrics())
+            {
+                var labels = metric.Labels;
+                var builder = new PrometheusMetricBuilder()
+                    .WithName(metric.MetricName)
+                    .WithDescription(metric.MetricDescription);
+
+                switch (metric.AggregationType)
+                {
+                    case AggregationType.DoubleSum:
```
Comment (MikeGoldsmith) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs]: You could omit this case if you're not interested in capturing it.

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                                var metricValueBuilder = builder.AddValue();
+                                metricValueBuilder = metricValueBuilder.WithValue(doubleValue);
+                                metricValueBuilder.WithLabel(label.Key, label.Value);
+                            }
+
+                            builder.Write(writer);
+                            break;
+                        }
+
+                    case AggregationType.LongSum:
```
Comment (MikeGoldsmith) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterExtensions.cs]: You could omit this case if you're not interested in capturing it.

File: src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        private readonly PrometheusExporter exporter;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PrometheusExporterMiddleware"/> class.
+        /// </summary>
+        /// <param name="next">The next middleware in the pipeline.</param>
+        /// <param name="exporter">The <see cref="PrometheusExporter"/> instance.</param>
+        public PrometheusExporterMiddleware(RequestDelegate next, PrometheusExporter exporter)
+        {
+            if (next is null)
```
Comment (MikeGoldsmith) [inline src/OpenTelemetry.Exporter.Prometheus/PrometheusExporterMiddleware.cs]: Could do inline null checking like you did for the PrometheusExporter.

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs [hunk 4]
```diff
@@ -56,11 +59,62 @@ public void PrometheusExporterTest1()
...
+        public async Task E2ETestMiddleware()
+        {
+            var promOptions = new PrometheusExporterOptions() { Url = "http://localhost:9184/metrics/" };
+            var promExporter = new PrometheusExporter(promOptions);
+            var simpleProcessor = new UngroupedBatcher(promExporter, new System.TimeSpan(100));
+
+            var builder = new WebHostBuilder()
+                .Configure(app =>
+                {
+                    app.Use((d) => new PrometheusExporterMiddleware(d, promExporter).InvokeAsync);
```
Comment (MikeGoldsmith) [inline test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs]: Would be nice to add a `UsePrometheus` extension for IApplicationBuilder.

**Comment by MikeGoldsmith** [comment]
@hananiel PS - you'll need to sign the CLA before we can accept this PR.

**Comment by cijothomas** [comment]
@hananiel LGTM.

**Comment by SergeyKanzhelev** [comment]
should this be moved out of Draft status? Or more changes are expected?

**Comment by MikeGoldsmith** [comment]
Looks ready to me 😄 👍

**Pull request merged**